src/main/java/dk/kb/util/webservice/OpenApiResource.java

```diff
@@ -195,7 +197,27 @@ private static String getReplacementForMatch(String yPath) {
         if (config == null){
             throw new IllegalStateException("Config must be initialized before using the class. See JavaDoc for OpenApiResource for further details.");
         }
-        return config.get(yPath).toString();
+
+        List<Object> result = config.getMultiple(yPath);
+
+        if (result.isEmpty()){
+            // I am not quite sure what we should return here. Is it too harsh to throw an exception? If not, which would be correct? Illegal Argument?
+            log.error("No entry has been found for yPath: '{}'.", yPath);
+            return yPath;
+        }
+        if (result.size() == 1){
+            return result.get(0).toString();
+        }
+
+        // If there are more than one entry, then the entries are combined to a specially formatted comma seperated string.
+        // All entries are seperated by ", " to make the openAPI generator see the input ["${config:yaml.string}"] as an
+        // actual array resolved as ["foo", "bar", "zoo"]
+        StringJoiner joiner = new StringJoiner("\", \"");
+        for (Object entry: result) {
+            joiner.add(entry.toString());
+        }
+
+        return joiner.toString();
```

Comment:
```Java
return Strings.join(result, "\", \"");
```
or maybe only half-quote Strings so that numbers & booleans can also be referenced? Not a common use case, so this might be over engineering.
```Java
        return result.get(0) instanceof String ?
                Strings.join(result, "\", \"") :
                Strings.join(result, ", ");
```

Reply:
For the enum lookup to work we need to enpasulate in "" as yaml syntax as this `enum: [${config:path}]`, wont go through the spec validation. So changing to your first suggestion